tools/launch.py

```diff
@@ -424,6 +432,59 @@ def wrap_cmd_with_extra_envvars(cmd: str, env_vars: list) -> str:
     env_vars = " ".join(env_vars)
     return wrap_cmd_with_local_envvars(cmd, env_vars)
 
+
+g_monitor_file = None
+g_group_id = 0
+
+def has_alive_servers(args):
+    """Check whether there exist alive servers and get available group ID if exist."""
```

Comment:
can you say more about how you check it?

Reply:
sure.